src/codegen/codegen_c_visitor.cpp

```diff
@@ -3794,7 +3794,7 @@ void CodegenCVisitor::print_net_move_call(const FunctionCall& node) {
     // artificial cells don't use spike buffering
     // clang-format off
     if (info.artificial_cell) {
-        printer->add_text(fmt::format("artcell_net_move(&{}, {}, nt->_t+", tqitem, pnt));
+        printer->add_text(fmt::format("artcell_net_move(&{}, {}, ", tqitem, pnt));
```

Comment:
Just as a reference and for my own education:

[invlfiresha.mod](https://github.com/neuronsimulator/tqperf/blob/add46f2899f1cde55032aa63cda54f3941e81155/modx/invlfiresha.mod#L137) has:

```python
 NET_RECEIVE (w) {
 VERBATIM
   nrnsha1_update((void*)_p_nrnsha1, &(t), sizeof(double));
 ENDVERBATIM
   m = M()
   t0 = t
   if (flag == 0) {
     ninput = ninput + 1
     m = m + w
     if (m > 1) {
       m = 0
       noutput = noutput + 1
       net_event(t)
     }
     net_move(t+firetime())
   }else{
     net_event(t)
     noutput = noutput + 1
     m = 0
     specify_invl()
     net_send(firetime(), 1)
   }
 }
```

and translated code fragment has:

```cpp
 /*VERBATIM*/
   nrnsha1_update((void*)_p_nrnsha1, &(t), sizeof(double));
  m = M ( _threadargs_ ) ;
    t0 = t ;
    if ( _lflag  == 0.0 ) {
      ninput = ninput + 1.0 ;
      m = m + _args[0] ;
      if ( m > 1.0 ) {
        m = 0.0 ;
        noutput = noutput + 1.0 ;
        net_event ( _pnt, t ) ;
        }
      artcell_net_move ( _tqitem, _pnt, t + firetime ( _threadargs_ ) ) ;
      }
    else {
      net_event ( _pnt, t ) ;
      noutput = noutput + 1.0 ;
      m = 0.0 ;
      specify_invl ( _threadargs_ ) ;
      artcell_net_send ( _tqitem, _weight_index, _pnt, t +  firetime ( _threadargs_ ) , 1.0 ) ;
      }
    }
```

the corresponding logic in [parsact.c](https://github.com/BlueBrain/mod2c/blob/8565d3c178a195a489fae0623d6338c2e92cd1e5/src/mod2c_core/parsact.c#L1134) is:

```cpp
		}else if (strcmp(SYM(qname)->name, "net_move") == 0) {
			if (artificial_cell) {
				replacstr(qname, "artcell_net_move");
			}
			if (blocktype == NETRECEIVE) {
                                acc_net_add(qname, qpar1, qexpr, qpar2, "2, _tqitem, -1, _ppvar[1*_STRIDE],", ", 0.");
				Insertstr(qpar1->next, "_tqitem, _pnt,");  // HERE net_move in NET_RECEIVE block don't get t + argument
			}else{
                             diag("net_move", " only allowed in NET_RECEIVE block");
			}
		}
...



// this is only relevant for non-artificial cells
void acc_net_add(Item* qname, Item* qpar1, Item* qexpr, Item* qpar2, const char* xarg1, const char* xarg2) {
  Item *q;
  if (artificial_cell) { return; }
  sprintf(buf, "\
\n#if NET_RECEIVE_BUFFERING\
\n    _net_send_buffering(_ml->_net_send_buffer, %s\
", xarg1);
  Insertstr(qname, buf);
  for (q = qpar1->next; q != qpar2; q = q->next) {
    if (q->itemtype == SYMBOL) {
	    insertsym(qname, SYM(q));
    }else if (q->itemtype == STRING) {
      insertstr(qname, STR(q));
    }else{
      diag("acc_net_add failed because item neither STRING nor SYMBOL");
    }
  }
  sprintf(buf, "%s);\n#else\n", xarg2);
  Insertstr(qname, buf);
  Insertstr(qpar2->next, "\n#endif\n");
}
```

Reply:
yes, I find it tricky to sometime need a t+ and sometime not. But , this is the past